lambdas/check-reverification-identity/src/main/java/uk/gov/di/ipv/core/checkreverificationidentity/CheckReverificationIdentityHandler.java

```diff
@@ -0,0 +1,186 @@
+            Gpg45ProfileEvaluator gpg45ProfileEvaluator,
+            VotMatcher votMatcher) {
+        this.configService = configService;
+        this.ipvSessionService = ipvSessionService;
+        this.clientSessionService = clientOAuthSessionDetailsService;
+        this.evcsService = evcsService;
+        this.userIdentityService = userIdentityService;
+        this.gpg45ProfileEvaluator = gpg45ProfileEvaluator;
+        this.votMatcher = votMatcher;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public CheckReverificationIdentityHandler() {
+        this(ConfigService.create());
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public CheckReverificationIdentityHandler(ConfigService configService) {
+        this.configService = ConfigService.create();
+        this.ipvSessionService = new IpvSessionService(configService);
+        this.clientSessionService = new ClientOAuthSessionDetailsService(configService);
+        this.evcsService = new EvcsService(configService);
+        this.userIdentityService = new UserIdentityService(configService);
+        this.gpg45ProfileEvaluator = new Gpg45ProfileEvaluator();
+        this.votMatcher = new VotMatcher(userIdentityService, gpg45ProfileEvaluator);
+    }
+
+    @Tracing
+    @Logging(clearState = true)
+    @Override
+    public Map<String, Object> handleRequest(JourneyRequest request, Context context) {
+        LogHelper.attachComponentId(configService);
+        configService.setFeatureSet(RequestHelper.getFeatureSet(request));
+
+        try {
+            var ipvSessionId = getIpvSessionId(request);
+            LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
+            var ipvSession = ipvSessionService.getIpvSession(ipvSessionId);
+            LogHelper.attachClientSessionIdToLogs(ipvSession.getClientOAuthSessionId());
+            var clientOAuthSession =
+                    clientSessionService.getClientOAuthSession(
+                            ipvSession.getClientOAuthSessionId());
+            LogHelper.attachClientIdToLogs(clientOAuthSession.getClientId());
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSession.getGovukSigninJourneyId());
+
+            var vcs =
+                    evcsService.getVerifiableCredentials(
+                            clientOAuthSession.getUserId(),
+                            clientOAuthSession.getEvcsAccessToken(),
+                            CURRENT);
+
+            var reverificationIdentityVot = getReverificationIdentityVot(vcs);
+            if (reverificationIdentityVot.isEmpty()) {
+                ipvSession.setFailureCode(NO_IDENTITY_AVAILABLE);
+                ipvSessionService.updateIpvSession(ipvSession);
+                return NOT_FOUND_RESPONSE;
+            }
+
+            if (GPG45.equals(reverificationIdentityVot.get().vot().getProfileType())) {
+                ipvSession.setTargetVot(reverificationIdentityVot.get().vot());
```

Comment:
Do we actually care about this? If the previous identity was a P2, do we _need_ the user to prove their identity to a P2 level? Would P1 give us enough confidence to let them use their MFA again?

Reply:
IMO this is kinda meaningless - the user doesn't prove to any GPG45 level (e.g. there's no fraud score!)

The only places it's used is to determine the CI threshold, and if we include F2F or KBV, then the scores being requested.

I wonder if we should approach this in a different way - either setting it to P2 automatically, or deliberately leaving it null and handling explicitly in all of those places.

(cc @DanCorderIPV who I think spotted this issue originally)

Reply:
Looking at the reverification journey, we'll never touch any of those things (checking if breaching CI threshold, or determining evidence required for F2F/KBV). So my inclination is to just leave it null and then handle.

Reply:
If you're confident that we'll never hit it, then that might be cleaner.

It does feel like we'd be leaving a little bit of a foot-gun if we change it later (e.g. to introduce KBVs), but I suppose we would hit it pretty quickly.

The remaining use-case that I'm not sure about is for the TICF CRI (i.e. `IpvSessionItem#getThresholdVot`)

Reply:
That sounds reasonable to me - I suspect we just set it to P2 originally because things broke if we didn't. If you're going to check those areas and make them not break then that should be OK.

Reply:
I'm gonna row back on my idea of leaving it as null. I failed to realise that we'll be checking for CIs as part of the return from a CRI, as well as in TICF, which requires a VOT.
We'll either need a mechanism to skip the CI check (which doesn't feel right), or decide on a threshold.

I don't want to put too much info about scores here, so I think lets talk about this offline.